### ISSUE TITLE
CLM vector-format history files and restart files pfts1d_wt  varnames

### ISSUE THREAD

**Issue by timhoar** [ESCOMP/ctsm#732]
As far as I can tell, the PFT weight variables names with the same long_name attribute could be made more consistent in the CLM vector-format history files and the CLM restart files.

In the CLM restart files, the variable is  pfts1d_wtxy
In the CLM vector  files, the variable is  pfts1d_wtgcell

ditto for pfts1d_wtlnd  and   pfts1d_wtlunit

        double pfts1d_wtxy(pft) ;
                pfts1d_wtxy:long_name = "pft weight relative to corresponding gridcell" ;
        double pfts1d_wtlnd(pft) ;
                pfts1d_wtlnd:long_name = "pft weight relative to corresponding landunit" ;
        double pfts1d_wtcol(pft) ;
                pfts1d_wtcol:long_name = "pft weight relative to corresponding column" ;

        double pfts1d_wtgcell(pft) ;
                pfts1d_wtgcell:long_name = "pft weight relative to corresponding gridcell" ;
        double pfts1d_wtlunit(pft) ;
                pfts1d_wtlunit:long_name = "pft weight relative to corresponding landunit" ;
        double pfts1d_wtcol(pft) ;
                pfts1d_wtcol:long_name = "pft weight relative to corresponding column" ;

Next time you get to code refactoring, please consider. It makes the accessor routines easier to support.

**Comment by billsacks** [comment]
Good point. This has bugged me before, though never enough to motivate me to actually do something about it.

I can see two main barriers to addressing this:

(1) How many people have scripts that depend on the current names, for either history or restart files?

(2) If we change the restart file names, we need to do this in a backwards-compatible manner. We have an easy mechanism for changing most restart file names in a backwards-compatible way (using a colon-delimited list of possible field names), which I'm pretty sure we could apply to these pfts1d_wt fields if those are the only fields you're concerned about. But I see some other inconsistencies, like `pfts1d_itype_veg` vs. `pfts1d_itypveg`, for which it would be a bit harder (but by no means impossible) to put in place backwards compatibility (because these fields get special treatment in init_interp - which is the main place where backwards compatibility of restart fields gets tricky).

If (1) weren't a concern, I'd be inclined to change the names on history files, because that avoids the backwards compatibility issues. But I suspect that more people have scripts that work with history files than with restart files, which would argue for changing the names on the restart files.

@ekluzek @olyson @swensosc - I'd be interested in hearing any of your thoughts on this.

**Comment by swensosc** [comment]
I use the vector output for the hillslope branch.  I have also used it for
crop related studies.  History files in both cases.  It might be worth
having both history and restart use a consistent three letter form that's
used in some other places: grc, lun, col, pft rather than things like
gcell, lunit, etc.

**Comment by olyson** [comment]
I use the vector output quite a bit, for producing pft-level plots and for processing urban data at the column and landunit level.  I also have a couple of scripts that process restart files, but these are used infrequently, and could be changed easily if we decide to change names on the restart files.

**Comment by timhoar** [comment]
Thanks for the discussion. This is a low-level nuisance that I thought I'd
bring up that might make it easier for 'future generations' of CLM users.
My workflow handles both restart and vector-based history, so I'm good.

Cheers --

Tim Hoar
Data Assimilation Research Section
National Center for Atmospheric Research
thoar@ucar.edu
303.497.1708

**Comment by billsacks** [comment]
@swensosc @olyson - do you feel it would be a pain (for you or others) if we changed the names on the history files?

**Comment by olyson** [comment]
For me personally, I think it is fine, as long as there is documentation about the mapping of each old variable to new variable, perhaps in the ChangeLog or on the wiki.  I think Danica and Will have versions of these vector processing scripts as well.  In Danica's case, I believe she might have made her scripts available to outside users. Just something to keep in mind.

**Comment by dlawrenncar** [comment]
My guess is that in the long run we would benefit from consistency and that
scripts could relatively straightforwardly be adjusted.
